### PR TITLE
[UI/UX:InstructorUI] Add exception reason for "My Late Days"

### LINKED ISSUE #10123: Add reason for excused absence into late days table
**What problem are you trying to solve with Submitty**
We recently added an optional string "reason" for each "excused absence extension"

We already display that string in rainbow grades.  But let's also add it to the late days table:
Screenshot 2024-01-22 at 9 47 57 PM

So in the "granted # of days extension" column, instead of just "1", let's make it for example "1 (illness)"
This is displayed in the ta grading interface in the "student information" panel.
And also when the student clicks on "my late days/extensions" from the left sidebar

### PR DESCRIPTION
**Pull request by Xiayucheng1212**
Closes #10123 

* [✅] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?

- No reason shown after the late days exception.
### What is the new behavior?
- Add reason for why these late days are acceptable.
- The excuse is shown in the parentheses.
截圖 2024-01-24 上午12 03 40

### CODE REVIEW

File: site/app/templates/LateDaysTablePlugin.twig [hunk 1]
```diff
@@ -43,7 +43,7 @@
             <td data-before-content="Event/Assignment Date">{{ late_day_info.getLateDayEventTime()|date('m/d/Y') }}</td>
             <td data-before-content="Max Late Days">{{ late_day_info.isLateDayUpdate() ? '' : (late_day_info.getAssignmentAllowedLateDays() ?? placeholder) }}</td>
             <td data-before-content="Days Submitted Late">{{ (late_day_info.getDaysLate() != 0) ? late_day_info.getDaysLate() : placeholder }}</td>
-            <td data-before-content="Days From Extensions">{{ (late_day_info.getLateDayException() != 0) ? late_day_info.getLateDayException() : placeholder }}</td>
+            <td data-before-content="Days From Extensions">{{ (late_day_info.getLateDayException() != 0) ? late_day_info.getLateDayException() != 0 ? late_day_info.getLateDayException() ~ ' (' ~ late_day_info.getReasonForException() ~ ') ' : placeholder }}</td>
```
Comment (DarthNyan) [inline site/app/templates/LateDaysTablePlugin.twig:46]:
```suggestion
            <td data-before-content="Days From Extensions">{{ (late_day_info.getLateDayException() != 0) ?  late_day_info.getLateDayException() ~ ' (' ~ late_day_info.getReasonForException() ~ ') ' : placeholder }}</td>
```

You shouldn't need to make this check a second time.

**Review by DarthNyan — CHANGES_REQUESTED** [review]

**Review by Xiayucheng1212 — COMMENTED** [review]
Thank you for pointing out.

**Review by DarthNyan — CHANGES_REQUESTED** [review]
Two changes, then this should be good to merge.
1. You've added a second space after the '?'. Please change that to a single space.
2. Can you edit the title to reflect that you're adding the exception reason to the "My Late Days" page? PR titles go in the release notes so it's helpful for readers if your title is specific about what page/s your changes affect.

**Comment by Xiayucheng1212** [comment]
@DarthNyan Thank you for your review. Both suggestions are applied, please check 👍

**Review by DarthNyan — APPROVED** [review]
Looks good to me

**Comment by Xiayucheng1212** [comment]
@DarthNyan  Hi, could you please tell me how to solve the Cypress checks for UI? I am not sure why this happens.

**Review by shailpatels — APPROVED** [review]
Thanks for the PR!

Not sure why CI was stuck but kicked off again via a merge so hopefully it goes through this time

**Review by williamjallen — APPROVED** [review]
Thanks for this contribution, and apologies for letting this one go unreviewed for such a long time!  This is now on track to get merged in the near future.

**Comment by bmcutler** [comment]
@Xiayucheng1212 Thanks for the contribution!

**Pull request merged**